### PR TITLE
feat: add tests and update multizone deployment

### PR DESCRIPTION
**Pull request by jor2**
### Description
Added an example for multi-zone deployment with secondary interface configured.

https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/issues/518

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by jor2** [comment]
WIP

File: examples/secondary-interface/main.tf [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  subnet_map = {
+    for subnet in module.slz_vpc.subnet_zone_list :
+    subnet.name => subnet
+  }
+}
+
+resource "ibm_is_subnet" "secondary_subnet" {
+  depends_on = [
+    module.slz_vpc
+  ]
```
Comment (ocofaigh) [inline examples/secondary-interface/main.tf:75]: why is the explicit depends on needed? There is already an implicit depends on with `module.slz_vpc.vpc_id`

File: examples/secondary-interface/variables.tf [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  type        = list(string)
+  description = "A list of access tags to apply to the VSI resources created by the module."
+  default     = []
+}
+
+variable "image_id" {
+  description = "Image ID used for VSI. Run 'ibmcloud is images' to find available images. Be aware that region is important for the image since the id's are different in each region."
+  type        = string
+  default     = "r006-1366d3e6-bf5b-49a0-b69a-8efd93cc225f"
+}
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:41]: dont expose this in the example - just use the modules default value

File: examples/secondary-interface/variables.tf [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+variable "machine_type" {
+  description = "VSI machine type"
+  type        = string
+  default     = "cx2-2x4"
+}
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:47]: dont expose this in the example - just use the modules default value

File: examples/secondary-interface/variables.tf [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+variable "create_security_group" {
+  description = "Create security group for VSI"
+  type        = string
+  default     = false
+}
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:53]: dont expose this in the example - just use hard code it in the example

File: examples/secondary-interface/variables.tf [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+          object({
+            type = number
+            code = number
+          })
+        )
+      })
+    )
+  })
+  default = null
+}
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:86]: dont expose this in the example - just use the modules default value

File: examples/secondary-interface/variables.tf [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+variable "user_data" {
+  description = "User data to initialize VSI deployment"
+  type        = string
+  default     = null
+}
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:92]: dont expose this in the example - just use the modules default value

File: examples/secondary-interface/variables.tf [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+variable "boot_volume_encryption_key" {
+  description = "CRN of boot volume encryption key"
+  type        = string
+  default     = null
+}
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:98]: dont expose this in the example - just use the modules default value

File: examples/secondary-interface/variables.tf [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+variable "vsi_per_subnet" {
+  description = "Number of VSI instances for each subnet"
+  type        = number
+  default     = 1
+}
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:104]: dont expose this in the example - just use the modules default value

File: examples/secondary-interface/variables.tf [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+  type        = string
+  description = "An existing ssh key name to use for this example, if unset a new ssh key will be created"
+  default     = null
+}
+
+variable "vpc_name" {
+  type        = string
+  description = "Name for VPC"
+  default     = "vpc"
+}
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:116]: dont expose this in the example - just use the prefix value for naming

File: examples/secondary-interface/variables.tf [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+variable "resource_group" {
+  type        = string
+  description = "An existing resource group name to use for this example, if unset a new resource group will be created"
+  default     = null
+}
+
+variable "region" {
+  description = "The region to which to deploy the VPC"
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:14]: deploy the VPC -> deploy all resources created by this example

File: examples/secondary-interface/variables.tf [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+variable "prefix" {
+  description = "The prefix that you would like to append to your resources"
+  type        = string
+  default     = "slz-vsi-sec-int"
+}
+
+variable "resource_tags" {
+  description = "List of Tags for the resource created"
+  type        = list(string)
+  default     = null
```
Comment (ocofaigh) [inline examples/secondary-interface/variables.tf:28]: use empty string for default value

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: examples/secondary-interface/variables.tf (hunk 3 above)
Comment (jor2) [reply examples/secondary-interface/variables.tf:47]: no default https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/blob/6e94f1edb696a4b9578af33c02ce70eaba8c1a9b/variables.tf#L80

File: examples/secondary-interface/variables.tf (hunk 4 above)
Comment (jor2) [reply examples/secondary-interface/variables.tf:53]: no default https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/blob/6e94f1edb696a4b9578af33c02ce70eaba8c1a9b/variables.tf#L119

File: examples/secondary-interface/variables.tf (hunk 6 above)
Comment (jor2) [reply examples/secondary-interface/variables.tf:92]: no default https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/blob/6e94f1edb696a4b9578af33c02ce70eaba8c1a9b/variables.tf#L90

File: examples/secondary-interface/variables.tf (hunk 8 above)
Comment (jor2) [reply examples/secondary-interface/variables.tf:104]: no default https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/blob/6e94f1edb696a4b9578af33c02ce70eaba8c1a9b/variables.tf#L85

**Comment by jor2** [comment]
> see comments

Updated PR.

**Review by ocofaigh — COMMENTED** [review]
@jor2 There will be some conflicts with https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/pull/536# so I'm thinking we maybe get that PR merged first. I'm also wondering if it makes sense to add the secondary interface example code into the new complete example being added in that PR?

**Comment by jor2** [comment]
> @jor2 There will be some conflicts with #536 so I'm thinking we maybe get that PR merged first. I'm also wondering if it makes sense to add the secondary interface example code into the new complete example being added in that PR?

makes sense, i'll add to complete example when its merged.

**Comment by ocofaigh** [comment]
@jor2 FYI https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/pull/536 is now merged

**Pull request closed without merging**